src/compute/mod.rs

```diff
@@ -129,8 +130,20 @@ fn compute_node_layout(
         }
     };
 
+    // Compute cache slot
+    let has_known_width = known_dimensions.width.is_some();
+    let has_known_height = known_dimensions.height.is_some();
+    let cache_slot = if has_known_width && has_known_height {
```

Comment:
The logic here is hard to follow, and seems likely to result in surprising bugs if altered unsuspectingly. Is there a clearer or more robust approach we can use? It feels like a match statement may help as a stopgap change.

Failing that, more comments.

Reply:
I think the logic itself is good (for the time being at least), but I agree that it could do with some explaining / documenting. I remember trying to understand the caching implementation when I first started working on Taffy, and it made no sense to me at all.